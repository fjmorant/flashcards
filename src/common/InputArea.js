import React from 'react';
import {
  View,
  TextInput,
  StyleSheet,
} from 'react-native';

const styles = StyleSheet.create({
  text: {
    flex: 1,
    borderColor: 'gray',
    borderWidth: 1,
    padding: 5,
  }
});

export default ({
  value,
  onChangeText,
  placeholder,
  ...props,
}) => (
  <TextInput
    {...props}
    multiline
    style={[styles.text, props.style]}
    onChangeText={onChangeText}
    placeholder={placeholder}
    value={value}/>
);
