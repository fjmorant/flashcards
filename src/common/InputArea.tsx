import * as React from 'react'
import {StyleSheet, TextInput, ViewStyle} from 'react-native'

const styles = StyleSheet.create({
  text: {
    flex: 1,
    borderColor: 'gray',
    borderWidth: 1,
    padding: 5,
  },
})

export interface IProps {
  value: string
  placeholder: string
  style: ViewStyle
  onChangeText: (text: string) => void
}

export default class InputArea extends React.Component<IProps> {
  public render() {
    const {value, onChangeText, placeholder, ...props} = this.props

    return (
      <TextInput
        {...props}
        multiline
        onChangeText={onChangeText}
        placeholder={placeholder}
        returnKeyType="next"
        style={[styles.text, props.style]}
        value={value}
      />
    )
  }
}
