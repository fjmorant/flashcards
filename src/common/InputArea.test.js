import 'react-native'
import React from 'react'
import InputArea from './InputArea'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<InputArea />).toJSON()
  expect(tree).toMatchSnapshot()
})
