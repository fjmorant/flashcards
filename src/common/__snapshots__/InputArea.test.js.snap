// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  multiline={true}
  onChangeText={undefined}
  placeholder={undefined}
  returnKeyType="next"
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
      },
      undefined,
    ]
  }
  value={undefined}
/>
`;
